src/main/java/org/mule/service/http/impl/service/client/InputCursorBodyGenerator.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.service.http.impl.service.client;
+
+import org.mule.runtime.api.streaming.Cursor;
+
+import com.ning.http.client.generators.InputStreamBodyGenerator;
+
+import java.io.IOException;
+
+import java.io.InputStream;
+
+public class InputCursorBodyGenerator extends InputStreamBodyGenerator {
```

Comment:
where is this used?

Reply:
The InputCursorBodyGenerator constructor receives an InputStream as parameter

Reply:
Why was this necessary?